Modules/TOF/src/TOFMatchedTracks.cxx

```diff
@@ -0,0 +1,242 @@
+#include "TOF/TOFMatchedTracks.h"
+#include <Framework/InputRecord.h>
+#include <Framework/InputRecordWalker.h>
+#include <Framework/InputSpec.h>
+#include "GlobalTrackingWorkflowHelpers/InputHelper.h"
+#include "ReconstructionDataFormats/TrackTPCITS.h"
+#include "ReconstructionDataFormats/MatchInfoTOF.h"
+#include "DataFormatsGlobalTracking/RecoContainerCreateTracksVariadic.h"
+
+namespace o2::quality_control_modules::tof
+{
+
+TOFMatchedTracks::~TOFMatchedTracks()
+{
+  for (int i = 0; i < trkType::SIZE; ++i) {
+    delete mInTracksPt[i];
+    delete mInTracksEta[i];
+    delete mMatchedTracksPt[i];
+    delete mMatchedTracksEta[i];
+    delete mEffPt[i];
+    delete mEffEta[i];
+  }
+}
+
+void TOFMatchedTracks::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  ILOG(Info, Support) << "initialize TOFMatchedTracks" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
+
+  // this is how to get access to custom parameters defined in the config file at qc.tasks.<task_name>.taskParameters
+  if (auto param = mCustomParameters.find("isMC"); param != mCustomParameters.end()) {
+    ILOG(Info, Devel) << "Custom parameter - isMC (= use of MC info): " << param->second << ENDM;
+    if (param->second == "true" || param->second == "True" || param->second == "TRUE") {
+      mUseMC = true;
+    }
+  }
+  if (auto param = mCustomParameters.find("verbose"); param != mCustomParameters.end()) {
+    ILOG(Info, Devel) << "Custom parameter - verbose (= verbose printouts): " << param->second << ENDM;
+    if (param->second == "true" || param->second == "True" || param->second == "TRUE") {
+      mVerbose = true;
+    }
+  }
+  if (auto param = mCustomParameters.find("GID"); param != mCustomParameters.end()) {
+    ILOG(Info, Devel) << "Custom parameter - GID (= sources by user): " << param->second << ENDM;
+    ILOG(Info, Devel) << "Allowed Sources = " << mAllowedSources << ENDM;
+    mSrc = mAllowedSources & GID::getSourcesMask(param->second);
+    ILOG(Info, Devel) << "Final requested sources = " << mSrc << ENDM;
+  }
+  mDataRequest = std::make_shared<o2::globaltracking::DataRequest>();
+  mDataRequest->requestTracks(mSrc, mUseMC);
+
+  if ((mSrc[GID::Source::TPCTOF] == 1 && mSrc[GID::Source::TPC] == 0) || (mSrc[GID::Source::TPCTOF] == 0 && mSrc[GID::Source::TPC] == 1)) {
+    ILOG(Fatal, Support) << "Check the requested sources: TPCTOF = " << mSrc[GID::Source::TPCTOF] << ", TPC = " << mSrc[GID::Source::TPC] << ENDM;
+  }
+
+  if ((mSrc[GID::Source::ITSTPCTOF] == 1 && mSrc[GID::Source::ITSTPC] == 0) || (mSrc[GID::Source::ITSTPCTOF] == 0 && mSrc[GID::Source::ITSTPC] == 1)) {
+    ILOG(Fatal, Support) << "Check the requested sources: ITSTPCTOF = " << mSrc[GID::Source::ITSTPCTOF] << ", ITSTPC = " << mSrc[GID::Source::ITSTPC] << ENDM;
+  }
+
+  for (int i = 0; i < trkType::SIZE; ++i) {
+    mInTracksPt[i] = new TH1D(Form("mInTracksPt_%d", i), Form("mInTracksPt (trkType %d); Pt; counts", i), 100, 0.f, 20.f);
```

Comment:
do you really need TH1D?

Reply:
I changed it, Initially it was TH1F, then I moved to TH1D thinking that maybe we want to integrate over longer times, but this is then a different way to implement, in the merger (which I have not yet tried myself). So let's use TH1F to start with, it makes sense.